vispy/visuals/image.py

```diff
@@ -654,23 +673,18 @@ def _compute_bounds(self, axis, view):
     def _build_color_transform(self):
         if self._data.ndim == 2 or self._data.shape[2] == 1:
             # luminance data
-            # bad_color handling is part of the colormap already, not needed here,
-            # but we put a noop functions to simplify indexing in the different cases
-            fbad = Function('float noop (float data) {return data;}')
             fclim = Function(self._func_templates['clim_float'])
             fgamma = Function(self._func_templates['gamma_float'])
             # NOTE: red_to_luminance only uses the red component, fancy internalformats
             #   may need to use the other components or a different function chain
             fun = FunctionChain(
-                None, [Function(self._func_templates['red_to_luminance']), fbad, fclim, fgamma, Function(self.cmap.glsl_map)]
+                None, [Function(self._func_templates['red_to_luminance']), fclim, fgamma, Function(self.cmap.glsl_map)]
```

Comment:
We need to be careful changing this function chain, because there are places (at least in this file) it is referenced directly by index. This breaks the clim setter, for example.

Reply:
ah yeah, I forgot to change it back. This PR actually goes back to how it was in main before I messed up #2659 (getting fixed in #2662)